src/Nethermind/Nethermind.Blockchain/BlockTree.cs

```diff
@@ -105,7 +105,11 @@ public partial class BlockTree : IBlockTree
         }
 
         public long BestKnownNumber { get; private set; }
+
         public long BestKnownBeaconNumber { get; private set; }
+
+        public ulong NetworkId => _specProvider.NetworkId;
+
         public ulong ChainId => _specProvider.ChainId;
```

Comment:
QQ: Any idea why is this in blocktree in the first place. Shouldnt its user inject specprovider on its own?

Reply:
It's quite convenient for a lot of tests at least. Can we count it as a bit ugly part of the code that is to big to refactor it for now?